From a7e6e645b640ed82ce6d6b2101e36694d5ef8cb3 Mon Sep 17 00:00:00 2001
From: Tian Yang <tiany@codeaurora.org>
Date: Tue, 29 Dec 2015 14:04:41 -0600
Subject: [PATCH] skbuff_debug: track alloc and free complete stacks

This adds support to record a complete stack trace for all
alloc's and free's of SKBs.

Change-Id: I81fc76240d49d18035e99c234abcb8e4b9cb14a5
Signed-off-by: Matthew McClintock <mmcclint@codeaurora.org>
Signed-off-by: Casey Chen <kexinc@codeaurora.org>
---
 include/linux/skbuff.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/include/linux/skbuff.h
+++ b/include/linux/skbuff.h
@@ -1068,7 +1068,9 @@ struct sk_buff {
 #endif
 
 #ifdef CONFIG_DEBUG_OBJECTS_SKBUFF
-	void			*free_addr;
+#define DEBUG_OBJECTS_SKBUFF_STACKSIZE	20
+	void			*free_addr[DEBUG_OBJECTS_SKBUFF_STACKSIZE];
+	void			*alloc_addr[DEBUG_OBJECTS_SKBUFF_STACKSIZE];
 #endif
 };
 
